Dockerfile

```diff
@@ -1,36 +1,57 @@
-# specify the node base image with your desired version node:<version>
-FROM node:12.16.3-slim
+# Build Stage
+FROM node:12.16.3-buster-slim AS umbrel-manager-builder
 
-# Install Tools
+# Install tools
 RUN apt-get update --no-install-recommends \
-    && apt-get install -y --no-install-recommends build-essential g++ \
-    && apt-get install -y --no-install-recommends git \
-    && apt-get install -y --no-install-recommends libltdl7 \
-    && apt-get install -y --no-install-recommends python \
-    && apt-get install -y --no-install-recommends rsync \
-    && apt-get install -y --no-install-recommends vim \
+    && apt-get install -y --no-install-recommends build-essential \
+    && apt-get install -y --no-install-recommends g++ \
+    && apt-get install -y --no-install-recommends make \
+    && apt-get install -y --no-install-recommends libssl-dev \
+    && apt-get install -y --no-install-recommends libffi-dev \
+    && apt-get install -y --no-install-recommends python3-dev \
+    && apt-get install -y --no-install-recommends python3-setuptools \
+    && apt-get install -y --no-install-recommends python3-wheel \
     && apt-get install -y --no-install-recommends python3 \
-    && apt-get install -y --no-install-recommends libssl-dev libffi-dev python3-dev python3-setuptools python3-wheel python3-pip \
+    && apt-get install -y --no-install-recommends python3-pip \
     && pip3 install -IU docker-compose \
     && ln -s /usr/local/bin/docker-compose /usr/bin/docker-compose \
-    && chmod +x /usr/local/bin/docker-compose \
-    && rm -rf /var/lib/apt/lists/*
+    && chmod +x /usr/local/bin/docker-compose
 
 # Create app directory
-WORKDIR /usr/src/app
+WORKDIR /app
 
-# Install app dependencies
-# A wildcard is used to ensure both package.json AND package-lock.json are copied
-# where available (npm@5+)
-COPY package.json ./
-COPY yarn.lock ./
+# Copy 'yarn.lock' and 'package.json'
+COPY yarn.lock package.json ./
 
-RUN yarn
-# If you are building your code for production
-# RUN npm install --only=production
+# Install dependencies
+RUN yarn install --production
 
-# Bundle app source
+# Copy project files and folders to the current working directory (i.e. '/app')
 COPY . .
 
+# Final image
+FROM node:12.16.3-buster-slim AS umbrel-manager
+
+# Install python3 and python3-pip (required for docker-compose)
+RUN apt-get update --no-install-recommends \
+    && apt-get install -y --no-install-recommends python3 \
+    && apt-get install -y --no-install-recommends python3-pip
```

Comment:
Are you sure `python3-pip` is required in the final build?

Reply:
Yeah, I tried without it and `docker-compose` wouldn't work as it's a `pip3` module.